testsuite/tests/basic/switch_opts.ml

```diff
@@ -0,0 +1,62 @@
+(* Test for optimisation of jump tables to arrays of constants *)
+
+let p = Printf.printf
+
+type test =
+  Test : 'b * 'a * ('b -> 'a) -> test
+
+type t = A | B | C
+
+(* These test functions need to have at least three cases.
+   Functions with fewer cases don't trigger the optimisation,
+   as they are compiled to if-then-else, not switch *)
+let testcases = [
+  Test (3, 3, function 1 -> 1 | 2 -> 2 | 3 -> 3 | _ -> 0);
+  Test (3, -3, function 1 -> 1 | 2 -> 2 | 3 -> -3 | _ -> 0);
+  Test (3, min_int, function 1 -> 1 | 2 -> 2 | 3 -> min_int | _ -> 0);
+  Test (3, max_int, function 1 -> 1 | 2 -> 2 | 3 -> max_int | _ -> 0);
+  Test (3, 3., function 1 -> 1. | 2 -> 2. | 3 -> 3. | _ -> 0.);
+  Test (3, ""^"c"^"", function 1 -> "a" | 2 -> "b" | 3 -> "c" | _ -> "");
```

Comment:
Hmm, maybe not---although it could be used to stop Flambda evaluating the string concatenation (which it might do in the future).  I'm not quite sure what to do here.  How important is it that the constants aren't shared?  That isn't clear to me.